feat(connectors): Slack channel creation (#1521)
camunda/camunda-platform-docs: docs/components/connectors/out-of-the-box-connectors/slack.md

![slack connector red properties](../img/connectors-slack-red-properties.png)

### Authentication

Set **Authentication** to your Slack OAuth token, which is stored as a secret. For example, 'secrets.SLACK_OAUTH_TOKEN'.

### Create channel

1. Set **Method** to `Create Channel`

> 2. Fill out the **New Channel Name**

Reviewer: ```suggestion
2. Set the **New Channel Name**
```

>    - Channel name can be up to 80 characters long and may contain lowercase letters, digits, and symbols `-` and `_`
>    - Can be provided as a FEEL expression

Reviewer: ```suggestion
   1. Channel name can be up to 80 characters long and may contain lowercase letters, digits, and symbols `-` and `_`
   2. Can be provided as a FEEL expression
```

![slack connector create channel](../img/connectors-slack-create-channel.png)

### Invite user to channel

To invite users to a channel, take the following steps:

1. Set **Method** to `Invite to Channel`.
2. Set the **Channel Name**:
   1. Channel name can be up to 80 characters and may contain lowercase letters, digits, and symbols `-` and `_`.
   2. This can be provided as a FEEL expression.
3. Set the **Users** as required:
   1. One single user name or email or id (for example: `@myUser` or `my.user@company.com` or `ABCDEF12345`).
   2. A comma separated list of users (for example: `@myUser, my.user@company.com, ABCDEF12345`).
   3. FEEL expression. In this case you can provide a valid list of strings (for example: `["@myUser", "my.user@company.com", "ABCDEF12345"]`).
   - Formats:
     - If the provided username starts with an `@` symbol, it will be handled as user name.
     - If the provided username is in an email format, it will be handled as an email.
     - If the provided username doesn't start with an `@`, and isn't an email, it will be handled as a user id.

>    - If you provide a channel name it will be omitted since it is not possible to invite a channel to an other channel.

Reviewer: ```suggestion
   - If you provide a channel name it will be omitted since it is not possible to invite a channel to another channel.
```

## Slack API response

The **Slack Connector** exposes the Slack API response as a [local variable](https://docs.camunda.io/docs/components/concepts/variables/#variable-scopes) called `response`.
Response contents are method-specific.

### Create channel

The following fields are available in the `response` variable after executing **Create Channel** method:

- **channel**:
  - **id**: channel ID
  - **name**: channel name

> Notice that **name** field can be subsequently used as an argument of **Post Message** method.

Reviewer: ```suggestion
Notice that the **name** field can be subsequently used as an argument of the **Post Message** method.
```

### Post message

> The following fields are available in the `response` variable after executing **Post Message** method.

Reviewer: ```suggestion
The following fields are available in the `response` variable after executing the **Post Message** method.
```

The **Slack Connector** communicates through a Slack app with a concrete Slack workspace. For example, when sending a Slack message, the message will be posted by the Slack app. For the **Slack Connector** to work, you need to perform the following steps:

1. [Create a Slack app](https://api.slack.com/apps).
2. [Request required scopes](https://api.slack.com/scopes) - The scopes represent what your app can and cannot do (for example, posting messages).

>    1. For **Create Channel** method to work, you need to grant at least the [`channels:manage`](https://api.slack.com/scopes/channels:manage) scope.
>    2. For **Post Message** method to work, at a minimum you need to grant the [`chat:write`](https://api.slack.com/scopes/chat:write) scope.

Reviewer: ```suggestion
   1. For the **Create Channel** method to work, you need to grant at least the [`channels:manage`](https://api.slack.com/scopes/channels:manage) scope.
   2. For the **Post Message** method to work, you need to grant at least the [`chat:write`](https://api.slack.com/scopes/chat:write) scope.
```
